main.go

```diff
@@ -142,6 +112,110 @@ func ProcessExportsSince(wh warehouse.Warehouse, since time.Time) (int, error) {
 	return len(exports), nil
 }
 
+// LoadFilesGrouped creates a single intermediate CSV file for all the export bundles on a given day.  It assumes the
+// day to be processed is the day from the first export bundle's Start value.  When all the bundles with that same day
+// have been written to the CSV file, it is loaded to the warehouse, and the function quits without attempting to
+// process remaining bundles (they'll get picked up on the next call to ProcessExportsSince)
+func LoadFilesGrouped(wh warehouse.Warehouse, fs *fullstory.Client, exports []fullstory.ExportMeta) (int, error) {
+	if len(exports) == 0 {
+		return 0, nil
+	}
+
+	log.Printf("Creating group file starting with bundle %d (start: %s)", exports[0].ID, exports[0].Start.UTC())
+	filename := filepath.Join(conf.TmpDir, fmt.Sprintf("%d-%s.csv", exports[0].ID, exports[0].Start.UTC().Format("20060102")))
+	mark := time.Now()
+	outfile, err := os.Create(filename)
+	if err != nil {
+		log.Printf("Failed to create tmp file: %s", err)
+		return 0, err
+	}
+	defer os.Remove(filename)
+	defer outfile.Close()
+	csvOut := csv.NewWriter(outfile)
+
+	var processedBundles []fullstory.ExportMeta
+	var totalRecords int
+	groupDay := exports[0].Start.UTC().Truncate(24 * time.Hour)
+	for _, e := range exports {
+		if !groupDay.Equal(e.Start.UTC().Truncate(24 * time.Hour)) {
+			break
+		}
+
+		recordCount, err := WriteBundleToCSV(fs, e.ID, csvOut, wh)
+		if err != nil {
+			return 0, err
+		}
+
+		log.Printf("Wrote bundle %d (%d records, start: %s, stop: %s)", e.ID, recordCount, e.Start.UTC(), e.Stop.UTC())
+		totalRecords += recordCount
+		processedBundles = append(processedBundles, e)
+	}
+
+	if err := wh.LoadToWarehouse(filename); err != nil {
+		log.Printf("Failed to load file '%s' to warehouse: %s", filename, err)
+		return 0, err
+	}
+
+	if err := wh.SaveSyncPoints(processedBundles); err != nil {
+		log.Printf("Failed to save sync points for bundles ending with %d: %s", processedBundles[len(processedBundles)].ID, err)
+		return 0, err
+	}
+
+	log.Printf("Processing of %d bundles (%d records) took %s", len(processedBundles), totalRecords,
+		time.Since(mark))
+
+	// return how many files were processed
+	return len(processedBundles), nil
+}
+
+func WriteBundleToCSV(fs *fullstory.Client, bundleID int, csvOut *csv.Writer, wh warehouse.Warehouse) (numRecords int, err error) {
+	stream, err := fs.ExportData(bundleID)
+	if err != nil {
+		log.Printf("Failed to fetch bundle %d: %s", bundleID, err)
+		return 0, err
+	}
+	defer stream.Close()
+
+	gzstream, err := gzip.NewReader(stream)
+	if err != nil {
+		log.Printf("Failed gzip reader: %s", err)
+		return 0, err
+	}
+
+	decoder := json.NewDecoder(gzstream)
+	decoder.UseNumber()
+
+	// skip array open delimiter
+	if _, err := decoder.Token(); err != nil {
+		log.Printf("Failed json decode of array token: %s", err)
```

Comment:
NIT Maybe mention which array token because the error message below is the same as well.